lightweight/calendareventsmodel/calendareventsmodel.cpp

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright (C) 2015 Jolla Ltd.
+ * Contact: Petri M. Gerdt <petri.gerdt@jollamobile.com>
+ *
+ * You may use this file under the terms of the BSD license as follows:
+ *
+ * "Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Nemo Mobile nor the names of its contributors
+ *     may be used to endorse or promote products derived from this
+ *     software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE."
+ */
+
+#include "calendareventsmodel.h"
+
+#include <QDBusInterface>
+#include <QDBusPendingReply>
+#include <QDBusPendingCallWatcher>
+#include <QDebug>
+#include <QFileSystemWatcher>
+#include <QTimer>
+#include <qqmlinfo.h>
+
+#include "calendardataserviceproxy.h"
+
+NemoCalendarEventsModel::NemoCalendarEventsModel(QObject *parent) :
+    QAbstractListModel(parent),
+    mProxy(0),
+    mWatcher(new QFileSystemWatcher(this)),
+    mUpdateDelayTimer(new QTimer(this)),
+    mFilterMode(FilterNone),
+    mContentType(ContentAll),
+    mEventLimit(1000),
+    mTotalCount(0)
+{
+    registerDataTypes();
+    mProxy = new CalendarDataServiceProxy("org.nemomobile.calendardataservice",
+                                          "/org/nemomobile/calendardataservice",
+                                          QDBusConnection::sessionBus(),
+                                          this);
+    connect(mProxy, SIGNAL(getEventsResult(EventDataList)),
+            this, SLOT(getEventsResult(EventDataList)));
```

Comment:
This method of separating query and result value to separate signal seems racy. Provided model has filters and all, so it can be very well assumed multiple models exist at the same time.

Now, if they simultaneously do updates, there's no way to know if the update signal was meant for this specific model. Event data on d-bus method return value wouldn't have that problem.